osconfig_tests/test_suites/patch/test_setup.go

```diff
@@ -29,58 +29,108 @@ type patchTestSetup struct {
 	machineType   string
 }
 
-var windowsSetup = &patchTestSetup{
-	assertTimeout: 15 * time.Minute,
-	startup: &compute.MetadataItems{
-		Key:   "windows-startup-script-ps1",
-		Value: &utils.InstallOSConfigGooGet,
-	},
-	machineType: "n1-standard-4",
-}
+var (
+	windowsRecordBoot = `
+$uri = 'http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/osconfig_tests/install_done'
+$old = Invoke-RestMethod -Method GET -Uri $uri -Headers @{"Metadata-Flavor" = "Google"}
+$new = $old+1
+Invoke-RestMethod -Method PUT -Uri $uri -Headers @{"Metadata-Flavor" = "Google"} -Body $new`
+	windowsStartup = utils.InstallOSConfigGooGet + windowsRecordBoot
 
-var aptSetup = &patchTestSetup{
-	assertTimeout: 5 * time.Minute,
-	startup: &compute.MetadataItems{
-		Key:   "startup-script",
-		Value: &utils.InstallOSConfigDeb,
-	},
-	machineType: "n1-standard-2",
-}
+	linuxRecordBoot = `
+uri=http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/osconfig_tests/boot_count
+old=$(curl $uri -H "Metadata-Flavor: Google" -f)
+new=$(($old + 1))
+curl -X PUT --data "${new}" $uri -H "Metadata-Flavor: Google"`
+	aptStartup = utils.InstallOSConfigDeb + linuxRecordBoot
+	el6Startup = utils.InstallOSConfigYumEL6 + linuxRecordBoot
+	el7Startup = utils.InstallOSConfigYumEL7 + linuxRecordBoot
```

Comment:
Is `gcloud` installed by default on these machines and would using that make this a little cleaner?

Reply:
Set guest attribute is not a gcloud command as it can only be run inside the guest